cfme/tests/cloud/test_delete_cloud_object.py

```diff
@@ -17,69 +18,74 @@ def pytest_generate_tests(metafunc):
 
 pytestmark = [pytest.mark.tier(2)]
 
+boolean = [True, False]
+
 
 @pytest.fixture(scope="module")
 def set_grid():
     sel.force_navigate("clouds_images")
     toolbar.select('Grid View')
 
 
-@pytest.fixture(scope="module")
-def set_grid_stack():
-    sel.force_navigate("clouds_stacks")
-    toolbar.select('Grid View')
-
-
 def reset():
+    # TODO replace this navigation with navmazing when cloud images supports it
     sel.force_navigate("clouds_images")
     toolbar.select('List View')
 
 
-def reset_grid_stack():
-    sel.force_navigate("clouds_stacks")
-    toolbar.select('List View')
+# TODO take generic object instead of stack when instance and image support navmazing destinations
+def refresh_and_wait(provider, stack):
+    provider.refresh_provider_relationships()
+    navigate_to(stack, 'All')
+    if not sel.is_displayed(Quadicon(stack.name, stack.quad_name)):
+        stack.wait_for_appear()
 
 
-def test_delete_instance(setup_provider, provider):
+@pytest.mark.parametrize('from_details', boolean, ids=['fromDetail', 'fromQuad'])
+def test_delete_instance(setup_provider, provider, from_details):
     """ Tests delete instance
 
     Metadata:
         test_flag: delete_object
     """
     instance_name = provider.data['remove_test']['instance']
     test_instance = VM.factory(instance_name, provider)
-    test_instance.delete()
+    delete_text = 'Remove Instance' if from_details else "Remove selected item"
+    test_instance.delete(from_details=from_details, btn_text=delete_text)
     test_instance.wait_for_delete()
     provider.refresh_provider_relationships()
     test_instance.wait_to_appear()
 
 
-def test_delete_image(setup_provider, provider, set_grid, request):
+@pytest.mark.parametrize('from_details', boolean, ids=['fromDetail', 'fromQuad'])
+def test_delete_image(setup_provider, provider, set_grid, request, from_details):
     """ Tests delete image
 
     Metadata:
         test_flag: delete_object
     """
+    # TODO as of 5.6+ clouds_images is no longer in the menu tree
+    # Refactor to navigate via clouds instances accordion
     image_name = provider.data['remove_test']['image']
     test_image = VM.factory(image_name, provider, template=True)
-    test_image.delete()
+    delete_text = 'Remove Image' if from_details else "Remove selected items"
+    test_image.delete(from_details=from_details, btn_text=delete_text)
     test_image.wait_for_delete()
     provider.refresh_provider_relationships()
     test_image.wait_to_appear()
     request.addfinalizer(reset)
 
 
 @pytest.mark.uncollectif(lambda: current_version() < "5.4")
-def test_delete_stack(setup_provider, provider, set_grid_stack, request):
+@pytest.mark.parametrize('from_details', boolean, ids=['fromDetail', 'fromQuad'])
```

Comment:
I would like to hear @psav input on this as well since I think it will be a very common pattern. 

Where would vary_from_quad_and_detail be defined?